FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download_Multithreaded/DemoTasks/S3DownloadMultithreadedHTTPExample.c

```diff
@@ -33,19 +33,24 @@
  * certificate, and then finally performs a TLS handshake with the HTTP server,
  * so that all communication is encrypted.
  *
- * Afterwards, a request and a response task are created, in addition to the
- * main demo task, along with two thread-safe queues (a request and response
- * queue), to demonstrate an asynchronous workflow for downloading an S3 file.
+ * Afterwards, two additional tasks are created (a request and response task),
+ * along with two thread-safe queues (a request and response queue), to
+ * demonstrate an asynchronous workflow for downloading an S3 file.
  *
- * The main task is responsible for sending HTTP requests to the server over the
- * transport interface, using the HTTP Client library API. It reads requests
- * from the request queue and places the corresponding server response on the
- * response queue. The request task is responsible for generating these
- * requests, specifying a byte range with each iteration in order to download
- * the S3 file in partial content responses. Finally, the response task logs and
- * evaluates the responses to these requests. The tasks run continuously until
- * the entire file is downloaded. If any request fails, an error code is
- * returned.
+ * There are three tasks to note in this demo:
+ * - prvHTTPDemoTask() is responsible for sending HTTP requests to the server
+ *   over the transport interface, using the HTTP Client library API. It reads
+ *   requests from the request queue and places corresponding server responses
+ *   on the response queue.
+ * - prvRequestTask() is responsible for generating request objects and placing
+ *   them on the request queue, specifying a byte range with each iteration in
+ *   order to download the S3 file in partial content responses.
+ * - prvResponseTask() logs and evaluates server responses to outgoing requests.
+ *   It reads responses from the response queue until the expected number of
+ *   responses have been received.
+ *
+ * The tasks run continuously until the entire file is downloaded. If any
```

Comment:
Is this true as the request task will end before the file is completely downloaded?

Reply:
Not entirely, I had meant it more in a general sense, but I will update it to be clearer. Thanks.